src/main/java/core/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package core.service.impl;
+
+import core.dao.ShoppingCartDao;
+import core.dao.TicketDao;
+import core.lib.Inject;
+import core.lib.Service;
+import core.model.MovieSession;
+import core.model.ShoppingCart;
+import core.model.Ticket;
+import core.model.User;
+import core.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+    
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticket = ticketDao.add(ticket);
```

Comment:
let's move this after getting shopping cart

Reply:
![image](https://user-images.githubusercontent.com/26012946/106882245-35bd2980-66e7-11eb-9dae-05d08e22b832.png)

Reply:
I put the getting the ShoppingCart to before adding the ticket